packages/editor/src/js/vue/components/save-block-modal/save-modal.js

```diff
@@ -0,0 +1,112 @@
+const Vue = require('vue/dist/vue.common');
+const { ModalComponent } = require('../modal/modalComponent');
+const styleHelper = require('../../utils/style/styleHelper');
+
+const SaveBlockModalComponent = Vue.component('SaveBlockModal', {
+  components: {
+    ModalComponent,
+  },
+  props: {
+    vm: { type: Object, default: () => ({}) },
+  },
+  data: () => ({
+    blockName: '',
+    blockCategory: '',
+    blockInformation: null,
+    style: styleHelper,
+    subscriptions: [],
+  }),
+  mounted() {
+    this.vm.toggleSaveBlockModal = this.handleToggleSaveBlockModalChange;
+  },
+  computed: {
+    disableSaveButton() {
+      return !this.blockName; // Disable the button if blockName is empty
+    },
+  },
+  methods: {
+    openModal() {
+      this.$refs.modalRef?.openModal();
+    },
+    closeModal() {
+      this.blockName = '';
+      this.blockCategory = '';
+      this.blockInformation = null;
+      this.$refs.modalRef?.closeModal();
+    },
+    handleToggleSaveBlockModalChange(value, data) {
+      if (value === true) {
+        this.openModal();
+        this.blockInformation = data;
+      } else {
+        this.closeModal();
+        this.blockInformation = null;
+      }
+    },
+    handleOnSubmit() {
+      // Logic to handle the submission of the block details
+      // This can be expanded upon based on your requirements
+      console.log(
+        'Block details submitted:',
+        this.blockName,
+        this.blockCategory,
+        this.blockInformation
+      );
```

Comment:
Is it a wanted behavior to keep console.log?

Reply:
Yes, to see a feedback when i click on submit button, but i think i will remove it